src/db/core.ts

```diff
@@ -258,6 +268,10 @@ export class SQLFragment<RunResult = pg.QueryResult['rows'], Constraint = never>
   }): SQLFragment<RunResult, Constraint> {
     const { literals = this.literals, expressions = this.expressions, ...overrideRest } = override ?? {};
     const copy = new SQLFragment<RunResult, Constraint>(literals, expressions);
+
+    // copy across relevant flags that aren’t part of the constructor
+    if (this.expectExactlyOne !== undefined) (copy as any).expectExactlyOne = this.expectExactlyOne;
```

Comment:
Can we not instead add `expectExactlyOne` to the set of properties copied across via `Object.assign`, just below?